gimme

```diff
@@ -81,13 +81,32 @@ _do_curl() {
 	exit 1
 }
 
+# _sha256sum "file"
+_sha256sum() {
+	if command -v sha256sum &>/dev/null; then
+		sha256sum "$@"
+	else
+		shasum -a 256 "$@"
+	fi
+}
+
 # _do_curls "file" "url" ["url"...]
 _do_curls() {
 	f="${1}"
 	shift
-	[[ ! -s "${f}" ]] || return 0
+	if _sha256sum -c "${f}.sha256" &>/dev/null; then
+		return 0
+	fi
 	for url in "${@}"; do
 		if _do_curl "${url}" "${f}"; then
+			if _do_curl "${url}.sha256" "${f}.sha256"; then
+				echo "  ${f}" >>"${f}.sha256"
+				if ! _sha256sum -c "${f}.sha256" &>/dev/null; then
+					: sha256sum failed for "${f}"
```

Comment:
Just use `echo >&2`?  `:` is not a "legacy comment", it's an alias for `true`, used idiomatically for side-effect of evaluating the parameters passed.  So these lines are `true blah blah`.  Which is okay for systems logging the commands, but less useful otherwise.

Reply:
Apologies, I meant that `:` was a "legacy comment" in the sense that it was a common way for humans to comment their sh prior to the existence of `# comment` [citation needed].

I'm also OK with sending to stderr 👍